### ISSUE TITLE
istioctl x waypoint delete does not clean up service account

### ISSUE THREAD

**Issue by SirNexus** [istio/istio#43687]
### Bug Description

After running `istioctl x waypoint apply --service-account bookinfo-productpage -n bookinfo`, a waypoint proxy is created for that service account as expected:

```
$ kubectl get pod -n bookinfo
bookinfo             bookinfo-productpage-istio-waypoint-66b44c865b-c8t9v   1/1     Running   0              23m
bookinfo             music-graphql-679b6f7bcf-sm86l                         2/2     Running   0              22h
bookinfo             productpage-v1-59cc5489df-vnzkr                        2/2     Running   0              22h
bookinfo             ratings-v1-b88b8f5f6-qw7vh                             2/2     Running   0              16h
bookinfo             reviews-v1-79bc78fd48-rbtcx                            2/2     Running   0              22h
bookinfo             reviews-v2-784dd568b6-52g8z                            2/2     Running   0              22h
bookinfo             users-grpc-7f6c79c59d-pcqh9                            2/2     Running   0              22h
```

As well as a service account:
```
$ kubectl get sa -n bookinfo
NAME                                  SECRETS   AGE
bookinfo-productpage                  0         22h
bookinfo-productpage-istio-waypoint   0         21h
bookinfo-ratings                      0         22h
bookinfo-reviews                      0         22h
default                               0         22h
```

However, when deleting that waypoint via istioctl, it does not clean up the service account:

```
$ istioctl x waypoint delete --service-account bookinfo-productpage -n bookinfo
waypoint bookinfo/default deleted
```

The waypoint pod is correctly deleted:
```
$ kubectl get pod -n bookinfo
bookinfo             music-graphql-679b6f7bcf-sm86l                         2/2     Running   0              22h
bookinfo             productpage-v1-59cc5489df-vnzkr                        2/2     Running   0              22h
bookinfo             ratings-v1-b88b8f5f6-qw7vh                             2/2     Running   0              16h
bookinfo             reviews-v1-79bc78fd48-rbtcx                            2/2     Running   0              22h
bookinfo             reviews-v2-784dd568b6-52g8z                            2/2     Running   0              22h
bookinfo             users-grpc-7f6c79c59d-pcqh9                            2/2     Running   0              22h
```

However the associated service account is not removed:
```
$ kubectl get sa -n bookinfo
NAME                                  SECRETS   AGE
bookinfo-productpage                  0         22h
bookinfo-productpage-istio-waypoint   0         21h
bookinfo-ratings                      0         22h
bookinfo-reviews                      0         22h
default                               0         22h
```

### Version

```prose
$ ./out/darwin_arm64/istioctl version

client version: 1.18-dev
control plane version: 1.18-alpha.54a87be0b45d834e26da33224cad0770b342d3ab
data plane version:  (3 proxies), 1.18-alpha.54a87be0b45d834e26da33224cad0770b342d3ab (5 proxies)
```

Istioctl dev commit: 76c45169acca08dad4599171265a7ae0dab2ea40

```
$ kubectl version --short
Flag --short has been deprecated, and will be removed in the future. The --short output will become the default.
Client Version: v1.24.5
Kustomize Version: v4.5.4
Server Version: v1.24.9
```

**Comment by GregHanson** [comment]
This is unfortunately a bug in upstream Kubernetes and we are tracking the fix.  Related issues:
 - https://github.com/kubernetes/kubernetes/issues/115511
 - https://github.com/kubernetes/kubernetes/issues/115459
 - https://github.com/kubernetes/kubernetes/pull/115644

**Comment by hanxiaop** [comment]
This has been fixed in upstream, and I think it'd be available in k8s 1.28.
